consensus/currency.go

```diff
@@ -128,3 +129,9 @@ func (c Currency) MarshalJSON() ([]byte, error) {
 func (c *Currency) UnmarshalJSON(b []byte) error {
 	return c.i.UnmarshalJSON(b)
 }
+
+// Scan implements the fmt.Scanner interface, allowing Currency values to be
+// scanned from text.
+func (c *Currency) Scan(s fmt.ScanState, ch rune) error {
```

Comment:
Does this work? Can an entire currency fit into a single rune?

I guess I just don't know.